### DISCUSSION TITLE
Access to GitHub Actions NPM registry for hostedtoolcache

### DISCUSSION THREAD

**Discussion by frankzo** [community/community#25013] [Actions]
<p>Since they switched Windows hostedtoolcache provisioner to GitHub Actions NPM registry you need to provide a PAT token for that registry; which I dont have.  How to gain access? </p>

**Comment by Yanjingzhu** [comment]
<p>You could follow <a href="https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line" rel="nofollow noopener">this</a> document to create a personal access token , when select scopes for the PAT, you need to check packages read and write scopes. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/333636598406642975caae24b6d5e6d5c232a7c8.png" data-download-href="/uploads/short-url/7j2v1dC0sgsDHN7rsPHK0cX4LFe.png?dl=1" title="npm PAT.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181088411-db663ec9-c853-43e5-9dea-4944f0e7e524.png" alt="npm PAT.png" data-base62-sha1="7j2v1dC0sgsDHN7rsPHK0cX4LFe" width="636" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/333636598406642975caae24b6d5e6d5c232a7c8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">npm PAT.png</span><span class="informations">695×546 27.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by frankzo** [comment]
<p>Thank this part I understand. I need to upload the packaged myself whereas I was looking for using the Microsoft actions github registry so I dont have to maintain my own registry (previously packages were downloaded from a public storage account )</p>

**Comment by Yanjingzhu** [comment]
<p>Do you mean you want a way to publish your NPM packages to Github Package Registry?</p>
<p>First, add a .npmrc file to the root of your repo with next content. </p>
<pre><code>registry=https://npm.pkg.github.com/{account}
//npm.pkg.github.com/:_authToken=${NODE_AUTH_TOKEN}
</code></pre>
<p>For example:</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181088416-b8384b76-0c4b-4653-ba77-30316a927cbd.png" alt="npmrc.png" data-base62-sha1="lRQHOgLGl5EtsDhf168P8gm5WoJ" width="500" height="245"></p>
<p>In your workflow, </p>
<p>when you run npm publish command, please add env NODE_AUTH_TOKEN with value of your PAT or GITHUB_TOKEN.</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181088417-e2b4ff3a-8774-46a2-90e9-5404103b1e6a.png" alt="npm publish.png" data-base62-sha1="53pTnDNzzQdYSYR2wmkm531gm4a" width="468" height="472"></p>

**Comment by frankzo** [comment]
<p>Hi thanks, but no. </p>
<p>I want to use the NPM registry that MS is using for the build of their hosted images. Otherwise I need to maintain that registry myself. </p>
<p>I clone the Repo virtualenvironments and want to use that to build my images. In order to do so I need to provide a PAT token for a registry that I do not have access to. </p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Do you mean you clone this <a href="https://github.com/actions/virtual-environments" rel="nofollow noopener">virtual-environment</a> repo? I checked the repo, it uses toolcache releated npm packages here: <a href="https://github.com/orgs/actions/packages" rel="nofollow noopener">https://github.com/orgs/actions/packages </a></p>
<p>As the packages are public packages , you could just try to use your own PAT to access it. </p>
<p>I tested in my account, after provide my own PAT, the package could be downloaded. </p>
<p><img src="https://user-images.githubusercontent.com/24806478/181088421-5d92d9b4-fabf-414f-9702-5f824a5c3337.png" alt="actions packages.png" data-base62-sha1="gKogJwfVTGOCwTDzosIyhL5FMeo" width="612" height="282"> </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2cc98970cccde76a6f8ab7e18a1f34f6b0256087.png" data-download-href="/uploads/short-url/6ocNz9FUVUFD5gjQQd5gprCBmgT.png?dl=1" title="install actions packages.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181088422-f072ee6a-a4fe-48e0-aa6e-48e89decdc42.png" alt="install actions packages.png" data-base62-sha1="6ocNz9FUVUFD5gjQQd5gprCBmgT" width="690" height="293" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2cc98970cccde76a6f8ab7e18a1f34f6b0256087_2_690x293.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/2cc98970cccde76a6f8ab7e18a1f34f6b0256087_2_1035x439.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2cc98970cccde76a6f8ab7e18a1f34f6b0256087.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2cc98970cccde76a6f8ab7e18a1f34f6b0256087_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">install actions packages.png</span><span class="informations">1184×504 28.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Please kindly let me know whether this time I understand your scenario correctly. </p>
